sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala

```diff
@@ -325,6 +325,21 @@ class CachedTableSuite extends QueryTest with SQLTestUtils with SharedSQLContext
     assert(isExpectStorageLevel(rddId, Memory))
   }
 
+  test("SQL interface support storageLevel(Invalid StorageLevel)") {
+    val message = intercept[IllegalArgumentException] {
+      sql("CACHE TABLE testData OPTIONS('storageLevel' 'invalid_storage_level')")
+    }.getMessage
+    assert(message.contains("Invalid StorageLevel: INVALID_STORAGE_LEVEL"))
+  }
+
+  test("SQL interface support storageLevel(with LAZY)") {
+    sql("CACHE LAZY TABLE testData OPTIONS('storageLevel' 'disk_only')")
+    assertCached(spark.table("testData"))
+    val rddId = rddIdOf("testData")
+    sql("SELECT COUNT(*) FROM testData").collect()
+    assert(isExpectStorageLevel(rddId, Disk))
```

Comment:
Although the commit message is a revert, this is not a revert. We lost the previous test coverage. Is this intentional?
```
"Lazily cached in-memory table shouldn't be materialized eagerly"
```

Reply:
Yes, I think this test cast is mainly to assert that `storageLevel` works with `lazy`, not `lazy` itself. WDYT?

Reply:
Do you think the previously existing `lazy`-related test cases protect this new SQL syntax contribution from future regressions?